interfaces/Portalicious/src/app/components/query-table/query-table.component.ts

```diff
@@ -484,10 +489,31 @@ export class QueryTableComponent<TData extends { id: PropertyKey }, TContext> {
    */
   visibleColumns = model<QueryTableColumn<TData>[]>([]);
 
-  resetColumnVisibility() {
-    this.visibleColumns.set(
-      this.columns().filter((column) => !column.defaultHidden),
+  resetColumnVisibility(removeFromLocalStorage = false) {
+    if (removeFromLocalStorage) {
+      localStorage.removeItem(this.selectedColumnsStateKey());
+    }
+
+    const storedSelectedColumns = localStorage.getItem(
+      this.selectedColumnsStateKey(),
     );
+    if (storedSelectedColumns) {
+      const interpretedColumns = JSON.parse(
+        storedSelectedColumns,
+      ) as QueryTableColumn<TData>[];
+      console.log(
+        '🚀 ~ QueryTableComponent<TData ~ resetColumnVisibility ~ interpretedColumns:',
+        interpretedColumns,
+      );
+      const matchedColumns = interpretedColumns
+        .map((column) => this.columns().find((c) => c.field === column.field))
+        .filter(Boolean) as QueryTableColumn<TData>[];
```

Comment:
```suggestion
        .filter((column) => !!this.columns().find((c) => c.field === column.field))  as QueryTableColumn<TData>[];
```

I don't think we need a double loop here